[RFC] Add USBG_VERSION preprocessor symbol

johnkeeping opened the pull request:
This is an RFC because I'm not sure it's the best way to extract a version number for use in a header (I looked at how some other projects do this and some use the header as the definitive source of truth for the version and do not include a package version in `configure.ac` but that would require more intrusive changes).

kopasiak commented on Makefile.am:
```diff
@@ -12,7 +12,7 @@ endif
 EXTRA_DIST = doxygen.cfg
 library_includedir=$(includedir)/usbg
-library_include_HEADERS = include/usbg/usbg.h
+library_include_HEADERS = include/usbg/usbg.h include/usbg/usbg_version.h
```
You add a new header so you will need to also update spec file (packaging/libusbgx.spec)

johnkeeping commented:
As far as I can see, libusb only defines a single `LIBUSB_API_VERSION` which is defined as:

```
 * Internally, LIBUSB_API_VERSION is defined as follows:
 * (libusb major << 24) | (libusb minor << 16) | (16 bit incremental)
```

I have updated this PR to use the same format as that and added a comment to the header to document the format.  I'm not averse to adding separate values for major, minor and patch level but I think they're less useful than a single combined value that can easily be checked in client code.

I've also updated the `.spec` file.

kopasiak commented:
I've added a macro to avoid those HEX numbers and renamed USBG_API_VERSION to USBG_VERSION. Please let me know if it works for you

johnkeeping commented:
Yes, that version works.

Are you likely to tag a new release soon?  Once this functionality is merged it would be helpful to bump the version number.

kopasiak commented:
Merged.
I hope to make a release in next week when we are sure that C++ issues has been fixed.
Thank you:)
